File: 백준/Silver/2559. 수열/수열.cc
#include <iostream>
#include <vector>
using namespace std;

int main() {
	int n, k, t;
	cin >> n >> k;

	vector<int> sum(n+1);
	for (int i = 1; i <=n; i++) {
		cin >> t;
		sum[i] = t + sum[i - 1];
	}

	int ans = sum[k];
	for (int i = k; i <= n; i++) {//n-k개
		ans = max(ans, sum[i] - sum[i - k]);
	}
	cout << ans;

	return 0;
}

File: 백준/Silver/2559. 수열/수열.py
n,k=map(int, input().split())
t=list(map(int, input().split()))

Sum=[0]*(n+1) #합 배열 초기화
for i in range(1,n+1):
    Sum[i]=t[i-1]+Sum[i-1]

ans = Sum[k]
for j in range(k, n+1):
    ans = max(ans, Sum[j]-Sum[j-k]) #최대값 찾기
print(ans)